R/combined.R

```diff
@@ -286,17 +293,40 @@ combined_aggregate_data <- function(data) {
 }
 
 
-combined_aggregate_rt <- function(rt, samples) {
+aggregate_severity <- function(severity, samples) {
+
+  severity <- switch_levels(severity)
+  hosp_names <- grep("hfr", names(severity), value = TRUE)
+  inf_names <- grep("hfr", names(severity), invert = TRUE, value = TRUE)
+
+  admissions <- switch_levels(severity[c("date", "step", hosp_names)])
+  infections <- switch_levels(severity[inf_names])
+
+
+  admissions <- combined_aggregate_rt(admissions, samples,
+                                      weight = "admitted_inc")
+
+  infections <- combined_aggregate_rt(infections, samples,
+                                      weight = "infections_inc")
+
+  severity <- list(
+    admissions = admissions,
+    infections = infections
+  )
+}
```

Comment:
I think we want this to work similarly to combined_aggregate_rt - it takes the severity object as it is currently and adds in severity$england and severity$uk (if we have nations included), then outputs that

Reply:
problem is we have different weights there

Reply:
more so, as we add severity disaggregated arrays (e.g. by variant, etc) both objects will get bigger with the ones weighted by admissions and by infections, respectively

Reply:
It only makes sense to weight IFR and IHR by infections, and HFR by admissions